server/src/main/java/org/phantazm/server/EquipmentFeature.java

```diff
@@ -215,6 +215,7 @@ private static void registerElementClasses(@NotNull ContextManager contextManage
         contextManager.registerElementClass(PotionShotHandler.class);
         contextManager.registerElementClass(SlowDownShotHandler.class);
         contextManager.registerElementClass(SoundShotHandler.class);
+        contextManager.registerElementClass(ZombiesPlayerMapStatsShotHandler.class);
```

Comment:
i think that gun stats shouldn't require a special shot handler to work, maybe bake this into Gun itself?

Reply:
then gun has a hard dependency on a ZombiesPlayer[MapStats]

Reply:
how is this a problem?

Reply:
I was thinking about using guns for non players but it might not be a problem

Reply:
just check if the shooter is a Player and don't record stats otherwise?

Reply:
I need an instance of a stats object though

Reply:
could be made Nullable, and you get the stats object through a ZombiesPlayer (idk if the last part is currently possible but it probably should be)

Reply:
alternatively, use Minestom's event system and record the statistics elsewhere (could be useful to avoid cluttering up Gun)